src/printing.ml

```diff
@@ -13,21 +14,27 @@ let print_array f sep fin fmt a =
 let string_of_name name =
   match name with
     | Names.Anonymous -> "anonymous"
-    | Names.Name n -> Names.string_of_id n
+    | Names.Name n -> Names.Id.to_string n
 
 let print_kn fmt kn =
-  fprintf fmt "%s" (Names.string_of_con (Names.constant_of_kn kn))
+  fprintf fmt "%s" (Names.string_of_con (Names.Constant.make1 kn))
+
+let coq_init_constant s =
+  Universes.constr_of_global @@
+    Coqlib.gen_reference_in_modules "RecursiveDefinition" Coqlib.init_modules s
+
+let coq_or = coq_init_constant "or"
 
 let rec print_constr fmt c =
   let f = print_constr in
   match kind_of_term c with
-  | _ when c = build_coq_False () -> fprintf fmt "False"
-  | _ when c = build_coq_True () -> fprintf fmt "True"
-  | _ when c = build_coq_not () -> fprintf fmt "Not"
-  | _ when c = build_coq_or () -> fprintf fmt "Or"
-  | _ when c = build_coq_and () -> fprintf fmt "And"
+  | _ when c = Universes.constr_of_global @@ Coqlib.build_coq_False () -> fprintf fmt "False"
```

Comment:
It's safe because all the constants you consider are monomorphic. Otherwise it would just raise an error. That said, I'd urge you to use instead the really correct form of this code, as it is going to be removed in 8.8. The correct way to write this is to use the `Globnames.is_global` function.